src/db/z_db.erl

```diff
@@ -197,6 +204,7 @@ return_connection(_C, _Context) ->
 
 %% @doc Apply function F with a connection as parameter. Make sure the
 %% connection is returned after usage.
+-spec with_connection(fun(), #context{}) -> any().
```

Comment:
The spec in line 207 uses a record, please define a type for the record and use that instead.